### PR TITLE
[SINT-1841] Dogfood Datadog SCA

### PR DESCRIPTION
**Pull request by juliendoutre**
### What does this PR do?

Add a new Gitlab CI job listing this repository dependencies with trivy and uploading them to Datadog. This will make it possible to use the SCA product for this repository (see https://app.datadoghq.com/ci/code-analysis?limit=100&offset=0).

### Motivation

@DataDog/software-integrity-and-trust partners with @DataDog/static-analysis to dogfood their SCA product and secure Datadog's supply chain.

### Additional Notes
I created and saved an app key in the AWS build-stable account under path: `ci.datadog-agent.datadog_app_key_org2` with appropriate permissions.  

### Describe how to test/QA your changes

Successful CI execution: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/439355805

Navigate to https://app.datadoghq.com/ci/code-analysis?limit=100&offset=0 and check `DataDog/datadog-agent` is visible in the list. Click it, and inspect its dependencies.

### CODE REVIEW

**Review by cedricvanrompay-datadog — COMMENTED** [review]
looks good to me

**Review by juli1 — APPROVED** [review]

File: .gitlab/software_composition_analysis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  stage: software_composition_analysis
+  tags:
+    - arch:amd64
+    - specific:true
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci-sbom:2024011006
+  when: always
+  allow_failure: true
+  dependencies: []
+  script:
+    - set +o xtrace  # mask secrets
```
[the remark this answers is not in the document]
Comment (juliendoutre) [reply .gitlab/software_composition_analysis.yml]: Added in [ae4f0b6](https://github.com/DataDog/datadog-agent/pull/22956/commits/ae4f0b6ffe0e3848d7cc555b82d969f575fedb6b).

File: .gitlab/software_composition_analysis.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  allow_failure: true
+  dependencies: []
+  script:
+    # disabling tracing to avoid leaking secrets
+    # see https://www.gnu.org/software/bash/manual/bash.html#The-Set-Builtin:
+    # "Using ‘+’ rather than ‘-’ causes these options to be turned off"
+    - set +o xtrace
+    - export DD_API_KEY=$(aws ssm get-parameter --region us-east-1 --name "ci.datadog-agent.datadog_api_key_org2" --with-decryption --query "Parameter.Value" --out text)
+    - export DD_APP_KEY=$(aws ssm get-parameter --region us-east-1 --name "ci.datadog-agent.datadog_app_key_org2" --with-decryption --query "Parameter.Value" --out text)
+    - export DATADOG_APP_KEY=$DD_APP_KEY
```
Comment (spencergilbert) [inline .gitlab/software_composition_analysis.yml:22]: **❓ question:** ‏Why are we exporting the App key variable "twice?"

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**